go/felix/ifacemonitor/iface_monitor.go

```diff
@@ -33,32 +42,33 @@ type InterfaceStateCallback func(ifaceName string, ifaceState State)
 type AddrStateCallback func(ifaceName string, addrs set.Set)
 
 type InterfaceMonitor struct {
+	netlinkStub  Netlink
 	upIfaces     set.Set
 	Callback     InterfaceStateCallback
 	AddrCallback AddrStateCallback
 	ifaceName    map[int]string
 	ifaceAddrs   map[int]set.Set
 }
 
-func New() *InterfaceMonitor {
+func New(netlinkStub Netlink) *InterfaceMonitor {
```

Comment:
I'd probably make the `New` constructor arg-less and have a separate constructor to pass in the shim.